### DISCUSSION TITLE
@apply and theme()

### DISCUSSION THREAD

**Discussion by kmarler18** [tailwindlabs/tailwindcss#10169] [Help] [answered]
Hey ya'll,

Two parter here.

The `theme` function in my project was throwing an error in Visual Studio Code.

![Screen Shot 2022-12-23 at 5 56 42 PM](https://user-images.githubusercontent.com/96091818/209411816-61da07dd-533d-4f7f-921c-31fa3eac1a7f.png)

I found putting the `theme` function's argument in quotes gets rid of the error message.

It looks like in the [documentation](https://tailwindcss.com/docs/functions-and-directives#theme) they don't use quotes when they use the `theme` function. 

I'm just wondering if I'm going to hurt anything in my project if I use the quotes. So far, the `theme` function seems to be doing its job, despite the quotes. 

Okay, part two.

I'm getting a warning message when using the `@apply `directive.

![Screen Shot 2022-12-23 at 6 26 56 PM](https://user-images.githubusercontent.com/96091818/209413124-bec024cd-45c8-4a5b-9fa4-ceda4c9727c4.png)

But, it still seems to work in my project. 

I wonder if these error/warning messages are b/c I'm using the 'Play CDN'? Am I good to keep on using the `@apply` directive and `theme` function in the manner I'm currently using them in?

Much love,
Kyle

**Comment by adamwathan** [comment, marked as answer]
This looks like you have a CSS linter running that is angry about our custom syntax. I would recommend just disabling the linter since it can't lint our stuff properly anyways. If you were using Tailwind with a build step, you could use our Tailwind CSS language mode for your CSS files but I don't think it will lint `<style>` blocks in HTML files, maybe there's a way to configure that though?

**Reply by kmarler18** [reply]
@adamwathan Thanks for taking the time for little old me Adam. I'm impressed. Merry Christmas!

**Comment by giniedp** [comment]
This is helpful. Tried to use the `theme` function in an angular project, but the compilation fails if the argument is not quoted.
